test/cpp/api/dataloader.cpp

```diff
@@ -1314,67 +1377,202 @@ class DummyChunkReader
   size_t chunk_size_;
 };
 
-TEST(DataTest, DataLoaderWithChunkSupportSingleWorker) {
-  const size_t kBatchSize = 13;
-  const size_t kNumChunks = 10;
+// This test tests the core function for iterate through a chunk dataset. It
+// contains test cases with different parameter combination. (For example,
+// different prefetch count, batch size and data loader worker count). It
+// verifies the return batches size and content when the order is deterministic.
+TEST(DataLoaderTest, ChunkDataSetGetBatch) {
```

Comment:
Perhaps, a better name would be ChunkDataSetGetBatchWithSingleWorker

Reply:
This test tests both single worker and multiple worker, so ChunkDataSetGetBatchWithSingleWorker would not cover all the case. I took a look at other exiting test case names, and found out not all of them are very specific, some just have the method name that is testing, for example, TEST(DataLoaderTest, Reset)

Reply:
Also, let's add bit of comment regarding this test. It is checking several things.

Reply:
I have added some explanation on line 1380. If you feel it is still not very clear, I can modify the comment.